Capabilities_Report_Design.md

```diff
@@ -0,0 +1,112 @@
+# Capabilities Report Design
+
+**Status**: WIP
+
+**Version**: Alpha
+
+**Author(s)**: Leon Wang (@leonwanghui)
+
+## Summary
+
+This proposal is drafted for building capabilities report mechanism for opensds storage backends, where a unified southbound interface is designed to standardize properties definition of opensds resource properties so as to provide some inputs for resource scheduling mechanism in opensds controller. 
+
+## Motivation
+
+As a unified storage reousrce management controller, it's indispensable to enable resource scheduling features for opensds, so the controller NEEDS to perceptive of all backend capabilities (such as capacities, configuration, advanced features and so forth). But currently what's standing in the way is that capabilities report methods in different storage systems are far different. For example, there is few property reported in LVM while some vendor sotrage systems can expose a lot of features. So we should design a standard capabilities report interface to integrate different storage drivers and establish capabilities discovery workflow to close the cycle of report mechanism.
+
+### Goals
+
+  * Design a unified API interface to collect capabilities.
+  * Design a workflow of capabilities discovery.
+
+### Non-Goals
+
+  * Integration to different storage drivers.
+  * Gap analysis of capabilities report feature among different backends.
+
+## Proposal
+
+Currently [Hotpot](https://github.com/opensds/opensds) project contains two services: ```osdslet``` service works as a controller to provide unified storage management, in the meantime ```osdsdock``` service runs on all storage nodes as a distributed agent. Since the capabilities report mechanism is a service from bottom to up, the service process should run on storage nodes and be triggered in polling mode. The concrete design would be introduced below.
+
+### Data model impact
+
+The proposal could imporve some properties in ```PoolSpec```, but considering the current format of ```ExtraSpec``` is map[string]interface{}, there would be no impact to the data model.
+
+Here is a sample JSON object of updated ```PoolSpec```:
+```json
+{
+  "id": "f4486139-78d5-462d-a7b9-fdaf6c797e1b",
+  "createdAt": "2018-01-15T12:19:45.505Z",
+  "updatedAt": "2018-01-15T12:19:45.505Z",
+  "name": "pool01",
+  "storageType": "block",
+  "description": "",
+  "status": "available",
+  "availabilityZone": "az01",
+  "totalCapacity": 1000,
+  "freeCapacity": 699,
+  "dockId": "ccac4f33-e603-425a-8813-371bbe10566e",
+  "extras": {
+    "DataStorageLineOfService": {
```

Comment:
Please remove "DataStorageLineOfService" and "DataProtectionLineOfService" because we won't include the name of LineOfService in the profile.  Just include properties such as "RecoveryTimeObjective" directly.